pandas/tests/indexes/test_base.py

```diff
@@ -790,32 +790,38 @@ def test_intersection_equal_sort_true(self):
         sorted_ = pd.Index(['a', 'b', 'c'])
         tm.assert_index_equal(idx.intersection(idx, sort=True), sorted_)
 
-    @pytest.mark.parametrize("sort", [None, False])
+    @pytest.mark.parametrize("sort", [None, True, False])
     def test_chained_union(self, sort):
         # Chained unions handles names correctly
         i1 = Index([1, 2], name='i1')
         i2 = Index([5, 6], name='i2')
         i3 = Index([3, 4], name='i3')
-        union = i1.union(i2.union(i3, sort=sort), sort=sort)
-        expected = i1.union(i2, sort=sort).union(i3, sort=sort)
+
+        warning = FutureWarning if sort is None else None
+        with tm.assert_produces_warning(warning):
```

Comment:
I think it would just be easier to move the warnings to a separate test since this will eventually be ripped out anyway